Auto update failing for yara

<br>Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the yara package, but it failed. 

Here's the output of the update script:

<details><summary>Output log</summary>
<pre lang="bash">

INFO: Updating yara [Current version: 4.2.0]...
INFO: package being updated to 4.2.1.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   133  100   133    0     0   2111      0 --:--:-- --:--:-- --:--:--  2111

100  960k    0  960k    0     0  2711k      0 --:--:-- --:--:-- --:--:-- 2711k
100 1255k    0 1255k    0     0  3083k      0 --:--:-- --:--:-- --:--:-- 5674k

You are about to commit these changes:

--------------------
diff --git a/packages/yara/build.sh b/packages/yara/build.sh
index 74e7f0a9b..e96de6c6b 100644
--- a/packages/yara/build.sh
+++ b/packages/yara/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://github.com/VirusTotal/yara
 TERMUX_PKG_DESCRIPTION="Tool aimed at helping malware researchers to identify and classify malware samples"
 TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=4.2.0
+TERMUX_PKG_VERSION="4.2.1"
 TERMUX_PKG_SRCURL=https://github.com/VirusTotal/yara/archive/v$TERMUX_PKG_VERSION.tar.gz
-TERMUX_PKG_SHA256=6f567d4e4b79a210cd57a820f59f19ee69b024188ef4645b1fc11488a4660951
+TERMUX_PKG_SHA256=f26d9c481e6789181431ac410665f6ba25d551c2948995f84c9e17df7a93731a
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_VERSION_REGEXP="\d+\.\d+\.\d+"
 TERMUX_PKG_DEPENDS="file, openssl"
--------------------

yara: update to 4.2.1


Not committing to Git!

INFO: Trying to build package.
Running container 'termux-package-builder' from image 'termux/package-builder'...
termux - building yara for arch aarch64...
Downloading https://packages-cf.termux.org/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  48825      0 --:--:-- --:--:-- --:--:-- 48825
Downloading https://packages-cf.termux.org/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   3559      0 --:--:-- --:--:-- --:--:--  3559
gpg: Signature made Tue 26 Apr 2022 06:41:14 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.com>" [ultimate]
Downloading https://packages-cf.termux.org/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13320  100 13320    0     0   149k      0 --:--:-- --:--:-- --:--:--  149k
Downloading https://packages-cf.termux.org/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   9800      0 --:--:-- --:--:-- --:--:--  9686
gpg: Signature made Fri 15 Apr 2022 10:57:29 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.com>" [ultimate]
Downloading https://packages-cf.termux.org/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0   137k      0 --:--:-- --:--:-- --:--:--  137k
Downloading https://packages-cf.termux.org/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   8768      0 --:--:-- --:--:-- --:--:--  8861
gpg: Signature made Tue 26 Apr 2022 02:59:02 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.com>" [ultimate]
Downloading dependency zlib@1.2.12 if necessary...
Found zlib in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/z/zlib/zlib_1.2.12_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 60072  100 60072    0     0   173k      0 --:--:-- --:--:-- --:--:--  173k
extracting zlib...
Downloading dependency file@5.41-1 if necessary...
Found file in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/f/file/file_5.41-1_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  317k  100  317k    0     0   771k      0 --:--:-- --:--:-- --:--:--  769k
extracting file...
Downloading dependency ca-certificates@1:2022.03.29 if necessary...
Found ca-certificates in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/c/ca-certificates/ca-certificates_1:2022.03.29_all.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  214k  100  214k    0     0   515k      0 --:--:-- --:--:-- --:--:--  515k
extracting ca-certificates...
Downloading dependency ca-certificates-java@1:2022.03.29 if necessary...
Found ca-certificates-java in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/c/ca-certificates-java/ca-certificates-java_1:2022.03.29_all.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   812  100   812    0     0   8371      0 --:--:-- --:--:-- --:--:--  8371
extracting ca-certificates-java...
Downloading dependency openssl@3.0.2 if necessary...
Found openssl in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/o/openssl/openssl_3.0.2_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 1544k  100 1544k    0     0  2952k      0 --:--:-- --:--:-- --:--:-- 2952k
extracting openssl...
Downloading dependency openssl-tool@3.0.2 if necessary...
Found openssl-tool in https://packages-cf.termux.org/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.org/apt/termux-main/pool/main/o/openssl-tool/openssl-tool_3.0.2_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  239k  100  239k    0     0   576k      0 --:--:-- --:--:-- --:--:--  576k
extracting openssl-tool...
Downloading https://github.com/VirusTotal/yara/archive/v4.2.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   133  100   133    0     0   1985      0 --:--:-- --:--:-- --:--:--  1985

100 1255k    0 1255k    0     0  4009k      0 --:--:-- --:--:-- --:--:-- 4009k
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:20: installing 'build-aux/ar-lib'
configure.ac:20: installing 'build-aux/compile'
configure.ac:38: installing 'build-aux/config.guess'
configure.ac:38: installing 'build-aux/config.sub'
configure.ac:8: installing 'build-aux/install-sh'
configure.ac:8: installing 'build-aux/missing'
Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
configure.ac: installing 'build-aux/ylwrap'
configure: WARNING: unrecognized options: --disable-rpath, --disable-rpath-hack, --disable-nls
checking whether make supports nested variables... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for aarch64-linux-android-strip... llvm-strip
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking for aarch64-linux-android-gcc... aarch64-linux-android-clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether aarch64-linux-android-clang accepts -g... yes
checking for aarch64-linux-android-clang option to accept ISO C89... none needed
checking whether aarch64-linux-android-clang understands -c and -o together... yes
checking dependency style of aarch64-linux-android-clang... none
checking for aarch64-linux-android-ar... llvm-ar
checking the archiver (llvm-ar) interface... ar
checking for aarch64-linux-android-gcc... (cached) aarch64-linux-android-clang
checking whether we are using the GNU C compiler... (cached) yes
checking whether aarch64-linux-android-clang accepts -g... (cached) yes
checking for aarch64-linux-android-clang option to accept ISO C89... (cached) none needed
checking whether aarch64-linux-android-clang understands -c and -o together... (cached) yes
checking dependency style of aarch64-linux-android-clang... (cached) none
checking for aarch64-linux-android-clang option to accept ISO C99... none needed
checking for flex... flex
checking lex output file root... lex.yy
checking lex library... none needed
checking whether yytext is a pointer... no
checking for bison... bison -y
checking for inline... inline
checking how to run the C preprocessor... aarch64-linux-android-cpp
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... (cached) no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking build system type... x86_64-pc-linux-gnu
checking host system type... aarch64-unknown-linux-android
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by aarch64-linux-android-clang... ld.lld
checking if the linker (ld.lld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... llvm-nm
checking the name lister (llvm-nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 3145728
checking how to convert x86_64-pc-linux-gnu file names to aarch64-unknown-linux-android format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld.lld option to reload object files... -r
checking for aarch64-linux-android-objdump... llvm-objdump
checking how to recognize dependent libraries... pass_all
checking for aarch64-linux-android-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for aarch64-linux-android-ar... (cached) llvm-ar
checking for archiver @FILE support... @
checking for aarch64-linux-android-strip... (cached) llvm-strip
checking for aarch64-linux-android-ranlib... llvm-ranlib
checking command to parse llvm-nm output from aarch64-linux-android-clang object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for aarch64-linux-android-mt... no
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if aarch64-linux-android-clang supports -fno-rtti -fno-exceptions... yes
checking for aarch64-linux-android-clang option to produce PIC... -fPIC -DPIC
checking if aarch64-linux-android-clang PIC flag -fPIC -DPIC works... yes
checking if aarch64-linux-android-clang static flag -static works... yes
checking if aarch64-linux-android-clang supports -c -o file.o... yes
checking if aarch64-linux-android-clang supports -c -o file.o... (cached) yes
checking whether the aarch64-linux-android-clang linker (ld.lld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... Android linker
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking whether to check for GCC pthread/shared inconsistencies... yes
checking whether -pthread is sufficient with -shared... yes
checking for isnan in -lm... yes
checking for log2 in -lm... yes
checking for strlcpy... yes
checking for strlcat... yes
checking for memmem... yes
checking for timegm... yes
checking for _mkgmtime... no
checking for clock_gettime... yes
checking stdbool.h usability... yes
checking stdbool.h presence... yes
checking for stdbool.h... yes
checking pkg-config is at least version 0.9.0... yes
checking openssl/md5.h usability... yes
checking openssl/md5.h presence... yes
checking for openssl/md5.h... yes
checking openssl/sha.h usability... yes
checking openssl/sha.h presence... yes
checking for openssl/sha.h... yes
checking openssl/asn1.h usability... yes
checking openssl/asn1.h presence... yes
checking for openssl/asn1.h... yes
checking openssl/crypto.h usability... yes
checking openssl/crypto.h presence... yes
checking for openssl/crypto.h... yes
checking openssl/bio.h usability... yes
checking openssl/bio.h presence... yes
checking for openssl/bio.h... yes
checking openssl/pkcs7.h usability... yes
checking openssl/pkcs7.h presence... yes
checking for openssl/pkcs7.h... yes
checking openssl/x509.h usability... yes
checking openssl/x509.h presence... yes
checking for openssl/x509.h... yes
checking openssl/safestack.h usability... yes
checking openssl/safestack.h presence... yes
checking for openssl/safestack.h... yes
checking for MD5_Init in -lcrypto... yes
checking for MD5_Update in -lcrypto... yes
checking for MD5_Final in -lcrypto... yes
checking for SHA256_Init in -lcrypto... yes
checking for SHA256_Update in -lcrypto... yes
checking for SHA256_Final in -lcrypto... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libyara/Makefile
config.status: creating libyara/yara.pc
config.status: executing depfiles commands
config.status: executing libtool commands
configure: WARNING: unrecognized options: --disable-rpath, --disable-rpath-hack, --disable-nls
Making all in libyara
make[1]: Entering directory '/home/builder/.termux-build/yara/build/libyara'
make  all-am
make[2]: Entering directory '/home/builder/.termux-build/yara/build/libyara'
  CC       grammar.lo
  CC       ahocorasick.lo
  CC       arena.lo
  CC       atoms.lo
  CC       base64.lo
  CC       bitmask.lo
  CC       compiler.lo
  CC       endian.lo
  CC       exec.lo
  CC       exefiles.lo
  CC       filemap.lo
  CC       hash.lo
  CC       hex_grammar.lo
  CC       hex_lexer.lo
  CC       lexer.lo
  CC       libyara.lo
  CC       mem.lo
  CC       modules.lo
  CC       notebook.lo
  CC       object.lo
  CC       parser.lo
  CC       proc.lo
  CC       re.lo
  CC       re_grammar.lo
  CC       re_lexer.lo
  CC       rules.lo
re_lexer.l:555:15: warning: result of comparison of constant -1 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
  if (text[1] == EOF || text[1] == 0)
      ~~~~~~~ ^  ~~~
re_lexer.l:562:17: warning: result of comparison of constant -1 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
    if (text[2] == EOF || text[2] == 0)
        ~~~~~~~ ^  ~~~
re_lexer.l:567:17: warning: result of comparison of constant -1 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
    if (text[3] == EOF || text[3] == 0)
        ~~~~~~~ ^  ~~~
3 warnings generated.
  CC       scan.lo
  CC       scanner.lo
  CC       sizedstr.lo
  CC       stack.lo
  CC       stopwatch.lo
  CC       strutils.lo
  CC       stream.lo
  CC       threading.lo
  CC       modules/tests/tests.lo
  CC       modules/elf/elf.lo
  CC       modules/math/math.lo
  CC       modules/time/time.lo
  CC       modules/pe/pe.lo
  CC       modules/pe/pe_utils.lo
  CC       modules/console/console.lo
  CC       modules/hash/hash.lo
  CC       modules/dotnet/dotnet.lo
  CC       proc/linux.lo
  CCLD     libyara.la
copying selected object files to avoid basename conflicts...
make[2]: Leaving directory '/home/builder/.termux-build/yara/build/libyara'
make[1]: Leaving directory '/home/builder/.termux-build/yara/build/libyara'
make[1]: Entering directory '/home/builder/.termux-build/yara/build'
  CC       cli/args.o
  CC       cli/common.o
  CC       cli/threading.o
  CC       cli/yara.o
  CC       cli/yarac.o
  CCLD     yarac
  CCLD     yara
ld: error: undefined symbol: libandroid_sem_close
>>> referenced by threading.c
>>>               cli/threading.o:(cli_semaphore_destroy)
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:1057: yara] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/yara/build'
make: *** [Makefile:1201: all-recursive] Error 1
ERROR: failed to build.
</pre>
</details>

Thanks!

<i>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/2226545151">2226545151</a><br>
Timestamp: 2022-04-26 12:20:12 UTC<br>
</i>

--- a/packages/yara/build.sh
+++ b/packages/yara/build.sh
@@ -2,15 +2,16 @@ TERMUX_PKG_HOMEPAGE=https://github.com/VirusTotal/yara
 TERMUX_PKG_DESCRIPTION="Tool aimed at helping malware researchers to identify and classify malware samples"
 TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=4.2.0
+TERMUX_PKG_VERSION=4.2.1
 TERMUX_PKG_SRCURL=https://github.com/VirusTotal/yara/archive/v$TERMUX_PKG_VERSION.tar.gz
-TERMUX_PKG_SHA256=6f567d4e4b79a210cd57a820f59f19ee69b024188ef4645b1fc11488a4660951
+TERMUX_PKG_SHA256=f26d9c481e6789181431ac410665f6ba25d551c2948995f84c9e17df7a93731a
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_VERSION_REGEXP="\d+\.\d+\.\d+"
-TERMUX_PKG_DEPENDS="file, openssl"
+TERMUX_PKG_DEPENDS="file, openssl, libandroid-posix-semaphore"
 TERMUX_PKG_BREAKS="yara-dev"
 TERMUX_PKG_REPLACES="yara-dev"
 
 termux_step_pre_configure() {
+	LDFLAGS+=" -landroid-posix-semaphore"
 	./bootstrap.sh
 }

